### DISCUSSION TITLE
app-1  | - pq: database "fider" does not exist

### DISCUSSION THREAD

**Discussion by vbresan** [getfider/fider#1174] [General]
I am trying to set up a self-hosted Fider on Google VPS. This is my first experience with both VPS and Docker. I was following the official instructions from [here](https://fider.io/docs/hosting-instance), using the same `docker-compose.yml` file with modified base URL and PostgreSQL user and password. 

Everything went OK until the very end. When I run `docker compose logs app` I get the following output:

```
WARN[0000] /var/fider/docker-compose.yml: `version` is obsolete 
app-1  | INFO [2024-06-30T11:47:51Z] [MIGRATE] Running migrations...
app-1  | INFO [2024-06-30T11:47:51Z] [MIGRATE] Found total of 71 migration files.
app-1  | ERROR [2024-06-30T11:47:51Z] [MIGRATE] Error Trace: 
app-1  | - failed to get last migration record (app/pkg/dbx/migrate.go:58)
app-1  | - dial tcp 172.18.0.3:5432: connect: connection refused
app-1  | INFO [2024-06-30T11:47:51Z] [MIGRATE] Running migrations...
app-1  | INFO [2024-06-30T11:47:51Z] [MIGRATE] Found total of 71 migration files.
app-1  | ERROR [2024-06-30T11:47:51Z] [MIGRATE] Error Trace: 
app-1  | - failed to get last migration record (app/pkg/dbx/migrate.go:58)
app-1  | - pq: database "fider" does not exist
...
app-1  | INFO [2024-06-30T12:19:51Z] [MIGRATE] Running migrations...
app-1  | INFO [2024-06-30T12:19:51Z] [MIGRATE] Found total of 71 migration files.
app-1  | ERROR [2024-06-30T12:19:51Z] [MIGRATE] Error Trace: 
app-1  | - failed to get last migration record (app/pkg/dbx/migrate.go:58)
app-1  | - pq: database "fider" does not exist
```

Any help is appreciated!

**Comment by vbresan** [comment]
Apparently, database needs to be created manually, which is not mentioned in the [documentation](https://fider.io/docs/hosting-instance). So I did the following:

1. `sudo docker ps` to list all containers and get the PostgreSQL container ID
2. `sudo docker exec -it <container_id> bash`
3. `psql -U <user>`
4. `CREATE DATABASE fider;`

After this, `sudo docker compose logs app` finally confirmed that http server is running.

**Comment by mattwoberts** [comment]
You shouldn't need to do that - the migrations should create the database for you. I just tested that to double check and it does this for me (I was testing a similar setup, using the linked to instructions and hosting in an Ubuntu EC2 instance)

**Reply by vbresan** [reply]
Well, I am trying to run Fider on Google VPS, also Ubuntu (24.04) and that is the error that I get. Do you have any other tips how to further inspect the problem? Does the error log give any clue?

**Reply by mattwoberts** [reply]
Did creating the database manually fix it for you, or do you still have issues after creating the database manually?

**Reply by vbresan** [reply]
No, creating the database manually didn't fix it for me. As you said earlier, I should not need to do that. Later I hit another wall when tenants table was missing ([issue 1171](https://github.com/getfider/fider/issues/1171)) and who knows what else.
